src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -0,0 +1,255 @@
+  *
+  * @param readNamePrefix the prefix to apply to all consensus read names
+  * @param readGroupId    the read group ID to apply to all created consensus reads
+  * @param minInputBaseQuality the minimum input base quality score to use a raw read's base
+  * @param errorRatePreUmi the estimated rate of errors in the DNA prior to attaching UMIs
+  * @param errorRatePostUmi the estimated rate of errors in the DNA post attaching UMIs
+  */
+class DuplexConsensusCaller(override val readNamePrefix: String,
+                            override val readGroupId: String = "A",
+                            val minInputBaseQuality: PhredScore = DuplexConsensusCaller.MinInputBaseQuality,
+                            val errorRatePreUmi: PhredScore = DuplexConsensusCaller.ErrorRatePreUmi,
+                            val errorRatePostUmi: PhredScore = DuplexConsensusCaller.ErrorRatePostUmi
+                           ) extends UmiConsensusCaller[DuplexConsensusRead] {
+
+  private val ssCaller =
+    new VanillaUmiConsensusCaller(readNamePrefix="x", options=new VanillaUmiConsensusCallerOptions(
+      errorRatePreUmi = this.errorRatePreUmi,
+      errorRatePostUmi = this.errorRatePostUmi,
+      minReads = 1,
+      minInputBaseQuality = this.minInputBaseQuality,
+      minConsensusBaseQuality = PhredScore.MinValue,
+      producePerBaseTags = true
+    )) {
+      /** Overriding here since we do this at the duplex level, and don't need to do it again for each SS. */
+      override protected[umi] def filterToMostCommonAlignment(recs: Seq[SAMRecord]): Seq[SAMRecord] = recs
+    }
+
+  /**
+    * Returns the MI tag minus the trailing suffix that identifies /A vs /B
+    */
+  override protected[umi] def sourceMoleculeId(rec: SAMRecord): String = {
+    val mi = rec.getStringAttribute(ConsensusTags.MolecularId)
+    require(mi != null, s"Read ${rec.getReadName} is missing it's ${ConsensusTags.MolecularId} tag.")
+    val index = mi.lastIndexOf('/')
+    require(index > 0, s"Read ${rec.getReadName}'s ${ConsensusTags} tag doesn't look like a duplex id: ${mi}")
+    mi.substring(0, index)
+  }
+
+  /**
+    * Takes in all the reads for a source molecule and, if possible, generates one or more
+    * output consensus reads as SAM records.
+    *
+    * @param recs the full set of source SAMRecords for a source molecule
+    * @return a seq of consensus SAM records, may be empty
+    */
+  override protected def consensusSamRecordsFromSamRecords(recs: Seq[SAMRecord]): Seq[SAMRecord] = {
+    // Group the reads by /A vs. /B and then sort so that the larger group is first, and then by suffix
+    val groups = recs.filter(_.getReadPairedFlag)
+      .groupBy(_.getStringAttribute(ConsensusTags.MolecularId))
+      .toSeq.sortWith {
+      case ((lhsMi, lhsRecs), (rhsMi, rhsRecs)) =>
+        if (lhsRecs.size > rhsRecs.size) true
+        else if (rhsRecs.size > lhsRecs.size) false
+        else lhsMi < rhsMi
+    }.map(_._2)
```

Comment:
I went back and forth on this.  At least in `GroupReadsByUmi` the /A and /B are largely artbirary.  Largely I just wanted some predictability for the tests.  I'll change it so that it just sorts by the id's for now.  That does mean that in filtering we have to be careful that the `a?` tags may have the lower values, but I think that's ok.